README.md

```diff
@@ -128,6 +128,19 @@ Check out detailed example below
 
 ![subdir](img/subdir_demo.png)
 
+### 2.3 Testing functions from notebook
+If you want to pass the functions from the notebook to the test case you need to enable `notebook_env`
+
```

Comment:
rather than introducing a second use case, why not systematically use `dill` so that lambdas are supported and the content writers do not need to wonder what they will be saving (which in time might be a source of bugs) in order to deal with that additional `notebook_env` parameter ?

if we keep a single use case, then the code only needs to deal with legacy content :

save :
- save as dill file
- remove pickle file if exists (no risk when the students run the code)

load :
- load from dill file
- fallback to load from pickle file

WDYT ?

Reply:
Dill can load any file created with pickle so I don't think we need the fallback. Are you suggesting we abstract away the saving of the notebook env?

Reply:
If you do this 

```
custom_ones = lambda x: np.ones(x)
```

you could not then pass this function alone

Reply:
I do not see the issue (I expect the code loading the dill files to deal with the package imports)

<img width="639" alt="Screenshot 2023-07-18 at 12 31 11" src="https://github.com/lewagon/nbresult/assets/34584935/e8b21658-9ded-429c-9089-3125bab0752b">

Reply:
try it in a test file rather than an interactive interpreter that is where I had the problem

Reply:
it works both with files and inside of a notebook (I restarted the kernel between save and load)

```
import dill
import numpy as np

custom_ones = lambda x: np.ones(x)

with open("a.dill", "wb") as file:
	dill.dump(custom_ones, file)
```

```
import dill
import numpy as np

with open("a.dill", "rb") as file:
	loaded = dill.load(file)

print(loaded([3, 1]))
```

Reply:
maybe the error was caused by the missing numpy import ?

Reply:
![Screenshot 2023-07-18 at 12 53 04](https://github.com/lewagon/nbresult/assets/74146610/a86b25aa-6d7a-43f1-a15e-89a4d78c9a6c)
![Screenshot 2023-07-18 at 12 53 00](https://github.com/lewagon/nbresult/assets/74146610/20ddf925-8448-4e86-81c6-ad17b63c7a00)

Reply:
I think it is specific to how pytest operates

Reply:
ok I get the issue with the numpy import when using the ChallengeResult class. Even if I manually import numpy the `np.ones` in the lambda fails to find numpy

stackoverflow also points to `dump_session`

Reply:
Haha that is very hacky 😓